### PR TITLE
Update convolution.hpp

### PR DESCRIPTION
**Pull request by InterTriplete2010**
Adding a description about how the input matrix of the CNN layer should be organized

### CODE REVIEW

File: src/mlpack/methods/ann/layer/convolution.hpp [hunk 1]
```diff
@@ -29,8 +29,26 @@ namespace ann /** Artificial Neural Network. */ {
...
+                *1, // Stride along width.
+                    *1, // Stride along height.
+                        *0, // Padding width.
+                            *0, // Padding height.
+                                *196, // Input width.
+                                    *14 // Input height.
+                                        *);
+ *By doing so, will recreate the original 196x14 matrix for each image (or Mel cepstral coefficients) that will be used as input for the 14 filters of this example.
+
+* @tparam ForwardConvolutionRule Convolution to perform forward process.
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/convolution.hpp:51]:
```suggestion
 * Example usage:
 * 
 * Suppose we want to pass a matrix M (2744x100) to a `Convolution` layer;
 * in this example, `M` was obtained from "flattening" 100 images (or Mel
 * cepstral coefficients, if we talk about speech, or whatever you like) of
 * dimension 196x14. In other words, the first 196 columns of each row of M
 * will be made of the 196 columns of the first row of each of the 100 images
 * (or Mel cepstral coefficients). Then the next 295 columns of M (196 - 393)
 * will be made of the 196 columns of the second row of the 100 images (or Mel
 * cepstral coefficients), etc.  Given that the size of our 2-D input images is
 * 196x14, the parameters for our `Convolution` layer will be something like
 * this:
 *
 * ```
 * Convolution<> c(1, // Number of input activation maps.
 *                 14, // Number of output activation maps.
 *                 3, // Filter width.
 *                 3, // Filter height.
 *                 1, // Stride along width.
 *                 1, // Stride along height.
 *                 0, // Padding width.
 *                 0, // Padding height.
 *                 196, // Input width.
 *                 14); // Input height.
 * ```
 *
 * This `Convolution<>` layer will treat each column of the input matrix `M` as  * a 2-D image (or object) of the original 196x14 size, using this as the input
 * for the 14 filters of this example.
 *
 * @tparam ForwardConvolutionRule Convolution to perform forward process.
```

Minor cleanups and wording changes, plus formatting to make it only 80 characters wide.  Feel free to modify further---it's just a suggestion. :)

**Review by rcurtin — COMMENTED** [review]
Hey @InterTriplete2010, thank you!  This looks great.  I have some small suggestions for formatting, etc., but I didn't change the content.  I think it is really nice to add an example like this.  Let me know what you think of my suggestion. :+1:

File: src/mlpack/methods/ann/layer/convolution.hpp (hunk 1 above)
Comment (InterTriplete2010) [reply src/mlpack/methods/ann/layer/convolution.hpp:51]: Hi @rcurtin, yes it looks much better with your changes 👍 Especially my sentence "I want my input to be 196x14 for, so my add will be something like this:" was poorly written (sorry, I didn't realize it until now). Thank you so much!!!

Comment (rcurtin) [reply src/mlpack/methods/ann/layer/convolution.hpp:51]: Cool, sounds good---I'll go ahead and commit the suggestion. :+1:  Thanks again for writing this up!

**Review by rcurtin — APPROVED** [review]
Looks great to me. :+1:

File: src/mlpack/methods/ann/layer/convolution.hpp [hunk 2]
```diff
@@ -29,6 +29,34 @@ namespace ann /** Artificial Neural Network. */ {
...
+ *                 3, // Filter height.
+ *                 1, // Stride along width.
+ *                 1, // Stride along height.
+ *                 0, // Padding width.
+ *                 0, // Padding height.
+ *                 196, // Input width.
+ *                 14); // Input height.
+ * ```
+ *
+ * This `Convolution<>` layer will treat each column of the input matrix `M` as  * a 2-D image (or object) of the original 196x14 size, using this as the input
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/convolution.hpp:58]:
```suggestion
 * This `Convolution<>` layer will treat each column of the input matrix `M` as
 * a 2-D image (or object) of the original 196x14 size, using this as the input
```

To make this line fit within our 80 character limit. See https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.

**Review by zoq — COMMENTED** [review]
Just one more minor style comment

File: src/mlpack/methods/ann/layer/convolution.hpp (hunk 2 above)
Comment (rcurtin) [reply src/mlpack/methods/ann/layer/convolution.hpp:58]: Oh, oops, I forgot to wrap that line somehow...

**Comment by rcurtin** [comment]
Thanks so much @InterTriplete2010! :+1:

**Pull request merged**